rush-plugins/rush-amazon-s3-build-cache-plugin/src/RushAmazonS3BuildCachePlugin.ts

```diff
@@ -17,14 +17,14 @@ const PLUGIN_NAME: string = 'AmazonS3BuildCachePlugin';
  */
 export interface IAmazonS3ConfigurationJson {
   /**
-   * The Amazon S3 region of the bucket to use for build cache (e.g. "us-east-1").
+   * The Amazon S3 endpoint of the bucket to use for build cache (e.g. "s3.us-east-2.amazonaws.com").
    */
-  s3Region: string;
+  s3Endpoint: string;
```

Comment:
To avoid making this a breaking change, would it make sense to add an additional option for `s3Endpoint` that is mutually exclusive with `s3Region`? If someone is using standard/public S3, they can just specify `s3Region`, and if someone is using a private or alternative provider, we can say they're in a more advanced scenario and they should instead specify the fully-qualified `s3Endpoint`.

Reply:
Good points!

`s3Region` will always be required, as it is needed for the signing the request.

`s3Bucket` and `s3Endpoint` can for sure be mutually exclusive, I can declare a `oneOf` specification in the build-cache.schema.json, and re-add the `_getHost` method I removed so we are backwards compatible and can support the most common simple use case.

Use case 1:
I just want to connect to `my-bucket` on S3:
```json
"s3Bucket": "my-bucket",
"s3Region": "eu-west-2",
```

Use case 2:
I just want to connect to `my-bucket` on S3, but I use the `s3Endpoint`:
```json
"s3Endpoint": "http://my-bucket.s3-eu-west-2.amazonaws.com",
"s3Region": "eu-west-2"
```

Use case 3:
I want to connect to `my-bucket` on `localhost`:
```json
"s3Endpoint": "http://localhost:9000/my-bucket",
"s3Region": "eu-west-2"
```

Use case 4:
I want to connect to `my-bucket` on `mydomain.com`:
```json
"s3Endpoint": "http://my-bucket.mydomain.com",
"s3Region": "eu-west-2"
```

What do you think?